Android streams implementation

kylelwm opened the pull request:
Created model folder with all model data inside
Implemented logout
Setup swipeRefreshLayout to list view for Android
Call and get stream data from server

kylelwm committed:
Refactor model data into seperate package and refactor previous model usages to reflect the change

kylelwm committed:
Fix Logout and remove TitleBar in Welcome screen

kylelwm committed:
Added default constructor to instantiate WorldScopeUser

kylelwm committed:
Updated XML layout and remove redundant fragments

kylelwm committed:
Set up SwipeRefreshLayout, hook up to ListView and setup dummy data into the list

kylelwm committed:
Updated method to instantiate Retrofit, created new API call getStreams and test getting stream in putFakeData()

kylelwm committed:
Included toString() methods for WorldScopeStream and WorldScopeViewStream

kylelwm committed:
Increase cache limit of tabs from 2 to 3

kylelwm committed:
Remove putFakeData() and add real data to streams, modified model to reflect changes

sharminelow commented on client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java:
```diff
@@ -1,31 +1,65 @@
         args.putInt(ARG_SECTION_NUMBER, sectionNumber);
         fragment.setArguments(args);
+        fragment.sectionNumber = sectionNumber;
+        fragment.sectionNumber = sectionNumber;
```
I think there's a copy paste error?

kylelwm replied:
Removed

sharminelow replied:
The duplicate error is still present

kylelwm replied:
Sorry, removed!

sharminelow commented on client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java:
```diff
@@ -35,8 +69,147 @@ public StreamRefreshListFragment() {
+        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                // Adds one more dummy when refresh
```
It doesn't add dummy already.

kylelwm replied:
Changed comment

kylelwm replied:
Rechanged comment

sharminelow commented on client/android/WorldScope/app/src/main/java/com/litmus/worldscope/model/WorldScopeCreatedStream.java:
```diff
@@ -0,0 +1,18 @@
+    private String streamLink;
+
+    public Date getEndedAt() {return endedAt;}
+    public String getStreamLink() {return streamLink;};
```
Typo, additional semi-colon

kylelwm replied:
Removed

sharminelow commented on client/android/WorldScope/app/src/main/java/com/litmus/worldscope/WorldScopeAPIService.java:
```diff
@@ -1,26 +1,45 @@
 
+import com.litmus.worldscope.model.WorldScopeUser;
+import com.litmus.worldscope.model.WorldScopeViewStream;
+import java.util.List;
```
the import statements can be separated by line feeds

kylelwm replied:
Seperated

sharminelow commented on client/android/WorldScope/app/src/main/java/com/litmus/worldscope/model/WorldScopeStream.java:
```diff
@@ -0,0 +1,83 @@
+    private String roomId;
+    private int totalStickers;
+    private int totalViewers;
+    private boolean live;
```
Just a heads-up, there will be another `int totalComments`

kylelwm replied:
Noted

sharminelow commented:
Done the review. You can look at the comments.

sharminelow commented:
@kylelwm I think the changes haven't been pushed.

kylelwm committed:
Remove GIF player dependency because it cause regression with JavaCV

kylelwm committed:
Refactor and reorder Gradle dependencies

kylelwm committed:
Reimplemented playing of GIF to use webview instead of regression library

kylelwm committed:
Added minimizing of app when back button is pressed in MainActivity

kylelwm committed:
Refactor FacebookLoginActivity's redirection to MainActivity to a method

kylelwm committed:
Account for case where no user is loaded into MainActivity

kylelwm committed:
Set up Android Instrumentation test environment for project

kylelwm committed:
Remove stub test and add tests for redirections

kylelwm committed:
Updated JavaCV version

kylelwm committed:
Refactored streaming functionality of StreamActivity into a fragment and fix camera issues such as rotation and release

kylelwm committed:
Changed theme of StreamActivity to fullscreen and allow opening of soft keyboard to resize Layout

kylelwm committed:
Edited streamVideoFragment to add preview only to its own View

kylelwm committed:
Created new Fragment that takes care of creating streams and is displayed on top of the preview

kylelwm committed:
Implemented the callback for the new StreamCreateFragment

kylelwm committed:
Remove misc Todos

kylelwm committed:
Added permission for Vibration

kylelwm committed:
Created new themes to support the StreamActivity requirements

kylelwm committed:
Refactored StreamActivity to act as a central hub for various fragments instead of just streaming videos

kylelwm committed:
Included fragments into the stream activity UI

kylelwm committed:
Fragment in charge of creating and stopping streams, also include UI elements for confirmation

kylelwm committed:
Removed old stream controls template and added new fragment for actual video streaming

kylelwm committed:
Created new API end point for creating a stream

kylelwm committed:
Refactor Date to a String for compatibility purposes

kylelwm committed:
Imported icon resources for Record button

kylelwm committed:
Created new StreamVideoControl fragment that handles user intervention during streaming

kylelwm committed:
Added logging for DashRendererBuilder

kylelwm committed:
Shorten success message for streaming

kylelwm committed:
Integrate clicking of stream to redirect into ViewActivity

kylelwm committed:
Make WorldScopeViewStream and WorldScopeStream parcelable so that it can be passed as Intent extra

kylelwm commented:
## New changes to original pull request

### Changelog

##### 1. Removed GIF player and use WebView to play GIF background instead

##### 2. Fix bug with pressing back button in main screen returning user to login screen

##### 3. Set up basic testing environment for Android client

##### 4. Wrote simple redirection tests for Android classes

##### 5. Updated JavaCV version and added filter for video to fix rotation issue

##### 6. Refactor and created several new fragments for streamer

###### \- Created a fragment for stream creation and termination

###### \- Created a fragment for stream controls

###### \- Created a fragment for actual streaming

##### 7. Added API end point for creating stream

##### 8. Redirect user to view screen when a stream is selected

##### 9. Refactor WorldScopeStream objects to be parcelable

kylelwm committed:
Fix comments in StreamRefreshListFragment and added Pretty Time library to format time

kylelwm committed:
Fix audio rate when streaming

kylelwm committed:
Fix null object reference error when App server is down

kylelwm committed:
Fix model to reflect correct date parsing

kylelwm committed:
Remove additional imports

kylelwm commented:
Please review =)

sharminelow commented:
Ok, good to proceed.

kylelwm commented:
Merged
